### ISSUE TITLE
Python3: The 'typing>=3.6.2' distribution was not found and is required by vim-vint

### ISSUE THREAD

**Issue by fszymanski** [Kuniwak/vint#238]
I get this exception:
```
Traceback (most recent call last):
  File "/usr/bin/vint", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3038, in <module>
    @_call_aside
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3022, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3051, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 657, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 971, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 857, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'typing>=3.6.2' distribution was not found and is required by vim-vint
```
`typing` is a standard library in Python 3.

**Comment by Kuniwak** [comment]
Oops, I'll fix it later.

Thank you for reporting!

**Comment by Kuniwak** [comment]
@fszymanski Hmmm, I could not reproduce it on MacOSX 10.12.6 with Python3.6 

```console
$ pip3 install vim-vint
Collecting vim-vint
  Using cached vim-vint-0.3.14.tar.gz
Collecting PyYAML~=3.11 (from vim-vint)
  Using cached PyYAML-3.12.tar.gz
Collecting ansicolor~=0.2.4 (from vim-vint)
  Using cached ansicolor-0.2.4-py2.py3-none-any.whl
Collecting chardet>=2.3.0 (from vim-vint)
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Requirement already satisfied: typing>=3.6.2 in /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages (from vim-vint)
Installing collected packages: PyYAML, ansicolor, chardet, vim-vint
  Running setup.py install for PyYAML ... done
  Running setup.py install for vim-vint ... done
Successfully installed PyYAML-3.12 ansicolor-0.2.4 chardet-3.0.4 vim-vint-0.3.14
```

Could you comment more information, please?

**Comment by AbsurdMagpieScrutinies** [comment]
Installed fine  4.13.0-rc5-686-pae #1 SMP Debian 4.13~rc5-1~exp1 (2017-08-14) i686 GNU/Linux


```

t > pip install vim-vint
Collecting vim-vint
  Downloading vim-vint-0.3.14.tar.gz (63kB)
    100% |████████████████████████████████| 71kB 1.3MB/s
Requirement already satisfied: PyYAML~=3.11 in /usr/lib/python2.7/dist-packages (from vim-vint)
Collecting ansicolor~=0.2.4 (from vim-vint)
  Downloading ansicolor-0.2.4-py2.py3-none-any.whl
Requirement already satisfied: chardet>=2.3.0 in /usr/lib/python2.7/dist-packages (from vim-vint)
Requirement already satisfied: enum34>=1.0.4 in /usr/lib/python2.7/dist-packages (from vim-vint)
Collecting pathlib==1.0.1 (from vim-vint)
  Downloading pathlib-1.0.1.tar.gz (49kB)
    100% |████████████████████████████████| 51kB 1.7MB/s
Collecting typing>=3.6.2 (from vim-vint)
  Downloading typing-3.6.2-py2-none-any.whl
Building wheels for collected packages: vim-vint, pathlib
  Running setup.py bdist_wheel for vim-vint ... done
  Stored in directory: /root/.cache/pip/wheels/b6/ff/de/a11959dfde6463d2e1bdc1742b09686b6885bcb9c954d55569
  Running setup.py bdist_wheel for pathlib ... done
  Stored in directory: /root/.cache/pip/wheels/2a/23/a5/d8803db5d631e9f391fe6defe982a238bf5483062eeb34e841
Successfully built vim-vint pathlib
Installing collected packages: ansicolor, pathlib, typing, vim-vint
Successfully installed ansicolor-0.2.4 pathlib-1.0.1 typing-3.6.2 vim-vint-0.3.14
```

**Comment by blueyed** [comment]
I was maybe only updated after @fszymanski tried it?

**Comment by Coacher** [comment]
Why always require typing when it's needed only with Python<3.5?

**Comment by blueyed** [comment]
Later/current setup.py improves/fixes this.
See #288.
I think this issue can be closed already.

**Comment by tyru** [comment]
This problem occurs on Travis CI.
https://travis-ci.org/tyru/open-browser.vim/builds/461401543

```
$ python --version
Python 3.4.6
$ pip --version
pip 9.0.1 from /home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages (python 3.4)

$ vint $VIM_FILES
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.6/bin/vint", line 7, in <module>
    from vint import main
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/vint/__init__.py", line 1, in <module>
    from vint.bootstrap import (
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/vint/bootstrap.py", line 5, in <module>
    from vint.linting.cli import CLI
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/vint/linting/cli.py", line 7, in <module>
    from vint.linting.linter import Linter
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/vint/linting/linter.py", line 5, in <module>
    from vint.encodings.decoder import EncodingDetectionError
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/vint/encodings/decoder.py", line 2, in <module>
    from typing import Dict, Any
ImportError: No module named 'typing'
The command "vint $VIM_FILES" exited with 1.
```

My setup is wrong?
https://github.com/tyru/open-browser.vim/blob/b900ff9d39bb36891704bd0fe76737ee3a7ac2b9/.travis.yml

**Comment by blueyed** [comment]
You are not using vint master, are you?
I've just merged https://github.com/Kuniwak/vint/pull/288, which should fix/address this.

**Comment by blueyed** [comment]
As a workaround you should install `typing` manually I guess (or use Python 3.6+).

**Comment by blueyed** [comment]
Should be fixed by #288, but only for 4.0+ / master.

**Comment by tyru** [comment]
@blueyed 
Ah, the latest version is not uploaded to PyPI, okay.
I'll fix .travis.yml if I have a time, or just wait until the next release.
Thanks!
